articles/quickstart/native/ios-swift-siwa/_includes/_login_centralized.md

```diff
@@ -0,0 +1,309 @@
+        switch(result) {
+        case .success(let credentials):
+
+            // NEW - store the user ID in the keychain
+            self.keychain.setString(appleIDCredential.user, forKey: "userId")
+
+            // NEW - store the credentials in the credentials manager
+            self.credentialsManager.store(credentials: credentials)
+
+        case .failure(let error):
+            print("Exchange Failed: \(error)")
+        }
+}
+```
+
+### Renew authentication state
+
+Add a function that tries to renew the user's login session.
+
+```swift
+func tryRenewAuth(_ callback: @escaping (Credentials?, Error?) -> ()) {
+    let provider = ASAuthorizationAppleIDProvider()
+        
+    // Try to fetch the user ID
+    guard let userID = keychain.string(forKey: "userId") else {
+        return callback(nil, nil)
+    }
+
+    // Check the Apple credential state
+    provider.getCredentialState(forUserID: userID) { state, error in
+        switch state {
+        case .authorized:
+            // Try to get credentials from the creds manager (ID token, Access Token, etc)
+            self.credentialsManager.credentials { error, credentials in
+                guard error == nil, let credentials = credentials else {
+                    return callback(nil, error)
+                }
+                
+                callback(credentials, error)
+            }
+            
+        default:
+            // User is not authorized - clear credentials
+            
+            self.keychain.deleteEntry(forKey: "userId")
+            self.credentialsManager.clear()
+            
+            callback(nil, error)
+        }
+    }
+}
+```
+
+:::note
+Calling `credentialsManager.credentials` _automatically renews_ the Access Token if it has expired, using the refresh token. This call should only execute if `getCredentialState` returns `authorized`, so the refresh token is only used by an authorized user. Otherwise, the credentials must be cleared and the login session thrown away.
+:::
+
+Finally, call this function from `viewDidLoad`. If no credentials are found, the user should be shown the login screen once more. Otherwise, they should continue on into the app:
+
+```swift
+tryRenewAuth { credentials, error in
+    guard error == nil, credentials == nil else {
+        print("Unable to renew auth: \(String(describing: error))")
+
+        // The user should be asked to log in again
+
+        return
+    }
+    
+    self.credentials = credentials
+
+    // Set up any post-login UI or segue here
+}
+```
+
+## Next Steps
+
+Now that you're able to use Sign In With Apple and exchange the authorization code for Auth0 credentials, you can use these credentials to perform various tasks:
+
+* Use the Access Token to [call APIs](/quickstart/native/ios-swift/04-calling-apis)
+* [Assign Roles to Users](/quickstart/native/ios-swift/05-authorization)
+* Use the ID token to [link accounts](/quickstart/native/ios-swift/07-linking-accounts)
```

Comment:
This should be the "access token" not "ID token"

Reply:
Thanks @tamigoodall!